### PR TITLE
Change tsk_table_collection_equals to take options.

### PR DESCRIPTION
**Pull request by jeromekelleher**
In favour of the ``tsk_table_collection_equals_with_options`` introduced in #897.

### CODE REVIEW

**Comment by AdminBot-tskit** [comment]
📖 Docs for this PR can be previewed [here](https://tskit-dev.github.io/tskit-docs/917/)

**Comment by jeromekelleher** [comment]
It would be good to get some feedback here at this point @benjeffery @petrelharp @mufernando. Basically, I've added an "options" argument to all the tables and made metadata comparisons optional now. I've also added an option for ignoring provenance timestamps, which will be handy when comparing simulation outputs.

Questions:

1. Do we want the ``TSK_CMP_NO_TS_METADATA`` option now? Seems like just saying ``TSK_CMP_NO_METADATA`` and not ignoring all the metadata is probably what we want, most of the time?
2. Should we also have an option to ignore the metadata schemas? If so, how does this interact with the ``TSK_CMP_NO_METADATA`` flag? Probably the simplest is the have TSK_CMP_NO_METADATA just ignore metadata and TSK_CMP_NO_METADATA_SCHEMA just ignore the schema?
2. What do we do at the Python level? Perhaps it would be better to have 

```python
def equals(self, other, metadata=True, provenance=True, provenance_timestamp=True):
```

rather than the current ``ignore_metadata=False`` or (worse) ``no_metadata=False``?

**Comment by petrelharp** [comment]
>     1. Do we want the `TSK_CMP_NO_TS_METADATA` option now? Seems like just saying `TSK_CMP_NO_METADATA` and not ignoring all the metadata is probably what we want, most of the time?

Well, in the use case of union'ing SLiM tree sequences, we do want to just ignore top-level metadata, so... yes?
 
>     2. Should we also have an option to ignore the metadata schemas? If so, how does this interact with the `TSK_CMP_NO_METADATA` flag? Probably the simplest is the have TSK_CMP_NO_METADATA just ignore metadata and TSK_CMP_NO_METADATA_SCHEMA just ignore the schema?

Well, ignoring metadata should ignore schemas, but I don't think we should ignore schemas if we compare metadata. It would maybe be useful to be able to ask to compare equivalency of schemas rather than equiality at the bytes level, but that's another can of worms (to do it in C anyhow). I'd say skip it unless we actually need it?

>     3. What do we do at the Python level? Perhaps it would be better to have
> 
> ```python
> def equals(self, other, metadata=True, provenance=True, provenance_timestamp=True):
> ```
> 
> rather than the current `ignore_metadata=False` or (worse) `no_metadata=False`?

Sounds good, plus maybe `top_level_metadata=metadata`?

**Comment by mufernando** [comment]
This all looks good to me. I agree that we need `TSK_CMP_NO_TS_METADATA` for top-level only metadata/schema.

**Comment by jeromekelleher** [comment]
I think this is ready for final review here. I ended up going back to the original "IGNORE_X" names, as the semantics got confusing when we changed back and forth between positive and negative in Python and C. "Ignore" is a bit longer, but I think it's quite clear what the flags go then.

Speaking of semantics - are we all happy that ``ignore_metadata`` implies ``ignore_ts_metadata``, and the latter has no effect when ``ignore_metadata=True``?

File: python/tests/test_tables.py [hunk 1]
```diff
@@ -2540,6 +2539,88 @@ def test_sequence_length_longer_than_edges(self):
         assert len(tree.parent_dict) == 0
 
 
+class TestEqualityOptions:
+    def test_equals_provenance(self):
+        t1 = msprime.simulate(10, random_seed=42).tables
+        t2 = msprime.simulate(10, random_seed=42).tables
```
Comment (benjeffery) [inline python/tests/test_tables.py:2545]: Windows is having trouble here as the resolution of `datetime.datetime.now()` is about 1ms(!!)
```suggestion
        t1 = msprime.simulate(10, random_seed=42).tables
        time.sleep(0.1)
        t2 = msprime.simulate(10, random_seed=42).tables
```

File: python/tskit/tables.py [hunk 2]
```diff
@@ -2146,14 +2180,58 @@ def __str__(self):
...
+        :param bool ignore_metadata: If True *all* metadata and metadata schemas
+            will be excluded from the comparison. This includes the top-level
+            tree sequence and constituent table metadata (default=False).
+        :param bool ignore_ts_metadata: If True the top-level tree sequence
+            metadata and metadata schemas will be excluded from the comparison.
+            If ``ignore_metadata`` is True, this parameter has no effect.
+        :param bool ignore_provenance: If True the provenance tables are
+            not included in the comparison.
+        :param bool ignore_timestamps: If True the provenance timestamp column
+            is ignored in the comparision.
```
Comment (benjeffery) [inline python/tskit/tables.py:2216]: Add a note that there is no effect if `ignore_provenance` it is there in the c doc and for `ignore_ts_metadata`

**Review by benjeffery — APPROVED** [review]
I agree with the semantics we have in this PR now. Just one minor doc suggestion and a broken windows test that is easily (if somewhat unsatisfyingly) fixed.

File: python/tskit/tables.py (hunk 2 above)
Comment (jeromekelleher) [reply python/tskit/tables.py:2216]: Good point, well spotted.

**Comment by mufernando** [comment]
LGTM!

**Comment by jeromekelleher** [comment]
OK, great, let's get this one in then.

**Pull request merged**